libreflect/src/stack_setup.c

```diff
@@ -96,16 +96,16 @@ void stack_setup(size_t *stack_base, int argc, char **argv, char **env, size_t *
 	dprint("  0x%08zx\n", stack_base[1 + ii]);
 
 	for (ii = 0; env[ii]; ii++) {
-		stack_base[1 + argc + ii] = (size_t)env[ii];
-		dprint("  0x%08zx\n", stack_base[1 + argc + ii]);
+		stack_base[2 + argc + ii] = (size_t)env[ii];
+		dprint("  0x%08zx\n", stack_base[2 + argc + ii]);
 	}
-	stack_base[1 + argc + ii] = 0;
-	dprint("  0x%08zx\n", stack_base[1 + argc + ii]);
+	stack_base[3 + argc + ii] = 0;
+	dprint("  0x%08zx\n", stack_base[3 + argc + ii]);
 
-	auxv_base = stack_base + 1 + argc + ii + 1;
+	auxv_base = stack_base + 1 + argc + ii + 3;
 
 	if(auxv) {
-		for (ii = 0; auxv[ii]; ii++) {
+		for (ii = 0; auxv[ii] || auxv[ii + 1]; ii++) {
```

Comment:
The auxiliary vector terminated solely by `AT_NULL`, it's value is not specified in the ABI. To be safe, this should increment by two and copy off in pairs, similar to how `AT_NULL` is set below:
```suggestion
		for (ii = 0; auxv[ii]; ii += 2) {
 			auxv_base[ii] = auxv[ii];
 			auxv_base[ii + 1] = auxv[ii + 1];
```